### PR TITLE
fix(Status Handler): Fix status handler bugs

### PR DESCRIPTION
**Pull request by ignacio-penas**
Fixed a bug where the Status notes were not updated when the System Manager was removed after an update. Also added timeout checks (1H timeout) to status report

Closes https://github.com/SixSq/tasklist/issues/3302

### CODE REVIEW

File: nuvlaedge/agent/common/status_handler.py [hunk 1]
```diff
@@ -40,17 +41,24 @@ def remove_module(self, module_name: str | StatusReport):
...
         def time_diff(date: datetime) -> int:
-            diff: timedelta = datetime.now() - date
-            return diff.seconds
+            diff: float = datetime.now().timestamp() - date.timestamp()
+            return int(diff)
+
+        to_remove = [name for name, report in self.module_reports.items() if time_diff(report.date) > self.STATUS_TIMEOUT]
+        if to_remove:
+            logger.info(f"Removing modules that have not reported in the last {self.STATUS_TIMEOUT} minutes: {', '.join(to_remove)}")
+            _ = [self.module_reports.pop(name) for name in to_remove]
```
Comment (schaubl) [inline nuvlaedge/agent/common/status_handler.py:58]: I think it should work without `_ = ` (not tested)
```suggestion
            [self.module_reports.pop(name) for name in to_remove]
```
did you try ?

**Review by schaubl — COMMENTED** [review]
Can we do the same for "_VPN Handler_" ? 
(I mean not adding it if VPN is disabled [`NUVLAEDGE_VPN_CLIENT_ENABLE`])

File: nuvlaedge/agent/common/status_handler.py (hunk 1 above)
Comment (ignacio-penas) [reply nuvlaedge/agent/common/status_handler.py:58]: Yes, that should work. I'll apply the change

**Comment by ignacio-penas** [comment]
Don't you think providing the information the the VPN is stopped could be helpful?

**Review by schaubl — APPROVED** [review]

**Pull request merged**